Newsletters: add 232 (2023-01-04) (#972)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2023-01-04-newsletter.md

---
title: 'Bitcoin Optech Newsletter #232'
permalink: /en/newsletters/2023/01/04/
name: 2023-01-04-newsletter
slug: 2023-01-04-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME:harding

## News

- **Software forks of Bitcoin Core:** last month saw the release of two
  patchsets on top of Bitcoin Core:

    - *Bitcoin Inquisition:* Anthony Towns [announced][towns bci] to the
      Bitcoin-Dev mailing list a version of [Bitcoin Inquisition][], a
      software fork of Bitcoin Core designed to be used on the default
      [signet][topic signet] for testing proposed soft forks and other
      significant protocol changes.  This version contains support for
      the [SIGHASH_ANYPREVOUT][topic sighash_anyprevout] and
      [OP_CHECKTEMPLATEVERIFY][topic op_checktemplateverify] proposals.
      Towns's email also includes additional information that will be
      useful to anyone participating in the signet tests.

    - *Full-RBF peering node:* Peter Todd [announced][todd rbf node] a

>       patch on top of Bitcoin Core 24.0.1 that sets a full-RBF service

Reviewer: Could link here as it could be interesting for readers to see the other service bits 

https://github.com/petertodd/bitcoin/commit/c15b8d70778238abfa751e4216a97140be6369af

      replacement would pay the original outputs at least the same
      amounts as the original transaction, ensuring the replacement
      mechanism couldn't be used to steal from the receiver of the
      original transaction.  Yuval Kogman [replied][kogman fss] with a
      link to an [earlier version][rbf-fss] of the same idea posted in
      2015 Peter Todd.  In a [subsequent][todd fss] reply, Todd
      described several ways in which the idea is much less preferable
      than opt-in or full RBF.

    - *Full-RBF motivation:* Anthony Towns [replied][towns rbfm] to a
      thread about the motivation for various groups to perform
      full-RBF.  Towns analyzes what economic rationality means---and
      does not mean---in the context of miner transaction selection.
      Miners optimizing for very short term profits would naturally
      prefer full-RBF.  However, Towns notes that miners who have made a
      long-term capital investment in mining equipment might instead
      prefer to optimize fee income over multiple blocks, and that might
      not always favor full-RBF.  He suggests three possible scenarios
      for consideration.

>         Towns also analyzes the phase, "Bitcoin is money for enemies"

Reviewer: I dont feel strongly on it, but this paragraph seems out of place in a typical Optech newsletter.

  using [opt-in RBF][topic rbf] in the GUI and when using the `bumpfee` RPC.

- [Bitcoin Core #24865][] allows a wallet backup to be restored on a
  node that has been pruned of older blocks as long as the node still
  has all of the blocks produced after the wallet was created.  The
  blocks are needed so that Bitcoin Core can scan them for any
  transactions affecting the wallet's balance.  Bitcoin Core is able to
  determine the age of the wallet because its backup contains the date
  the wallet was created.

- [Bitcoin Core #23319][] updates the `getrawtransaction` RPC to provide
  additional information if the `verbose` parameter is set to `2`.  The
  additional information includes the fee the transaction paid and
  information about each of the outputs from previous transactions
  ("prevouts") which are spent by being used as inputs to this
  transaction.  See [Newsletter #172][news172 prevout] for details about
  the method used to retrieve the information.

- [Bitcoin Core #26628][] begins rejecting RPC requests that include the
  same parameter name multiple times.  Previously, the daemon treated a

>   request with repeated parameters as if it only had the last of the repeated

Reviewer 2: ```suggestion
  request with repeated JSON keys as if it only had the last of the repeated
```

  parameters, e.g. `{"foo"="bar", "foo"="baz"}` was treated as
  `{"foo"="baz"}`.  Now the request will fail.  When using `bitcoin-cli`
  with named parameters, the behavior is unchanged: repeated parameters
  will not be rejected but only the last of the repeats will be sent.

- [Eclair #2464][] adds the ability to trigger an event when a remote
  peer becomes ready to process payments.  This is especially useful in
  the context of [async payments][topic async payments] where the local
  node temporarily holds a payment for a remote peer, waits for the peer
  to connect (or reconnect), and delivers the payment.

- [Eclair #2482][] allows sending payments using [blinded routes][topic
  rv routing], which are paths whose last several hops are chosen
  by the receiver.  The receiver uses onion encryption to obfuscate the

>   hop details and provides the encrypted data to the spender, along with

Reviewer: ```suggestion
  hop details and provide the encrypted data to the spender, along with
```

Reviewer 2: I'm not sure about your suggestion, @bitschmidty; I think the subject of the verb "provides" is the receiver (singular). Maybe it seems like "provide" is correct because "details" is plural?

  the identity of the first node along the blinded route.  The spender
  then constructs a payment path to that first node and includes the
  encrypted details for the operators of the last several nodes to
  decrypt and use to forward the payment to the receiver.  This allows
  the receiver to accept a payment without disclosing the identity of
  their node or channels to the spender, improving privacy.

- [LND #2208][] begins preferring different payment paths depending on
  the maximum capacity of a channel relative to the amount to be spent.
  As the amount to be sent approaches the capacity of a channel, that
  channel becomes less likely to be selected for a path.  This is
  broadly similar to pathfinding code already used in Core Lightning and
  LDK.

- [LDK #1738][] and [#1908][ldk #1908] provide additional features for handling
  [offers][topic offers].

- [Rust Bitcoin #1467][] adds methods for calculating the size in
  [weight units][] of transaction inputs and outputs.

> - [Rust Bitcoin #1330][] removes the `PackedLockTime` type, requiring
>   code instead use the almost-identical `absolute::LockTime` type.  A

Reviewer 3: I think if the code part of this sentence is dropped it reads easier

```suggestion
- [Rust Bitcoin #1330][] removes the `PackedLockTime` type, requiring
  code use of the almost-identical `absolute::LockTime` type.  A
```

Author: Changed to "downstream code".  I want to communicate that this isn't just an internal change, but a change that anyone depending on Rust Bitcoin for locktimes may need to account for (although hopefully it's something that will be caught by the linker when they next upgrade their dependencies, so this is just advance notice).

    - *Full-RBF peering node:* Peter Todd [announced][todd rbf node] a
      patch on top of Bitcoin Core 24.0.1 that sets a full-RBF service
      bit when it advertises its network address to other nodes,
      although only if the node is configured with `mempoolfullrbf`
      enabled.  Nodes running the patch also connect to up to four
      additional peers which had advertised that they support full-RBF.
      Peter Todd notes that [Bitcoin Knots][], another full node, also
      advertises the service bit, although it doesn't contain code to
      specifically peer with nodes advertising full-RBF support.

- **Continued RBF discussion:** in ongoing discussion about enabling
  [full-RBF][topic rbf] on mainnet, several parallel discussions were
  held last month on the mailing list:

    - *Full-RBF nodes:* Peter Todd probed full nodes which advertised
      that they were running Bitcoin Core 24.x and were accepting
      incoming connections on an IPv4 address.  He [found][todd probe]
      that about 17% relayed a full-RBF replacement: a transaction which
      replaced a transaction that did not contain the [BIP125][]

>       nSequence signal.  This indicates those nodes were running with

Author: (I really like this phrasing change, thanks!)
